cmd/kube-dns/app/options/options.go

```diff
@@ -95,9 +97,44 @@ func (m kubeMasterURLVar) Type() string {
 	return "string"
 }
 
+type federationsVar struct {
+	nameDomainMap map[string]string
+}
+
+func (fv federationsVar) Set(v string) error {
+	for _, val := range strings.Split(v, ",") {
+		splits := strings.SplitN(strings.TrimSpace(val), "=", 2)
+		name := strings.TrimSpace(splits[0])
+		domain := strings.TrimSpace(splits[1])
+		if !validation.IsDNS1123Label(name) {
+			return fmt.Errorf("%s not a valid federation name", name)
+		}
+		// The federation domain name need not strictly be domain names, we
+		// accept valid dns names with subdomain components.
+		if !validation.IsDNS1123Subdomain(domain) {
+			return fmt.Errorf("%s not a valid federation name", name)
+		}
+		fv.nameDomainMap[name] = domain
+	}
+	return nil
+}
+
+func (fv federationsVar) String() string {
+	var splits []string
+	for name, domain := range fv.nameDomainMap {
+		splits = append(splits, fmt.Sprintf("%s=%s", name, domain))
+	}
+	return strings.Join(splits, ",")
+}
+
+func (fv federationsVar) Type() string {
```

Comment:
This is to satisfy the pflag's `Value` interface so that we can pass it as an argument to fs.Var here - https://github.com/kubernetes/kubernetes/pull/25727/commits/3ad8ad1cd287055947d5a0669b3d1a61f19cbbd6#diff-0f7323379621678150800d4b5da58230R139